Add public example

vstrateman opened the pull request:
This PR:
-DRYs up client id and secret 
-adds patient read scope
-creates a public SMART launch example

bkaney commented:
I think we should update `examples/README.md`.

Also, in the .js file, could we call out the public workflow (and maybe how it's different that the confidential workflow)?

vstrateman commented:
@bkaney I added some description but wasn't sure how in depth we want to go with this. 
Most of the description is under the public-smart-ehr file, which was also renamed to better differentiate the public and confidential.

pdw207 commented:
@vstrateman Which EHR sandboxes have you tested this against?

Jammjammjamm commented on examples/confidential-smart-ehr/confidential-smart-ehr-launch.js:
```diff
@@ -0,0 +1,120 @@
+  saveUninitialized: true,
+}));
+
+
```
Extra whitespace.

Jammjammjamm commented on examples/README.md:
```diff
@@ -2,10 +2,10 @@
 
 This example provides two routes, `/launch` and `/callback`, through which an EHR
-may launch the SMART app within the EHR's provided launch context.
+may launch the SMART app within the EHR's provided launch context. This example demonstrates a confidential app, meaning that the application runs on a trusted server and can protect secret data, such as the CLIENT_SECRET variable.
```
Could you wrap the long lines in this file?

Jammjammjamm commented on examples/README.md:
```diff
@@ -23,6 +23,36 @@ send back a launch_context containing, among other things, an access token to
+The example provides two routes, `/launch` and `/callback`, through which an EHR
+may launch the SMART app within the EHR's provided launch context.
+
+To use:
```
Let's just refer to the instructions above as is done for the third example rather than repeating them here.
